File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int max = 0;
        
        for(int i=1;i<s.length();i++){
            String s1 = s.substring(0,i);
            String s2 = s.substring(i, s.length());
            int c1 = 0;
            for(int j=0;j<s1.length();j++){
                if(s1.charAt(j)=='0'){
                    c1++;
                }
            }
            int c2 = 0;
            for(int j=0;j<s2.length();j++){
                if(s2.charAt(j)=='1'){
                    c2++;
                }
            }
            int res = c1 + c2;
            if(max<res){
                max = res;
            }
        }
        return max;
    }
}

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        zeroCount = 0
        oneCount = 0
        max = 0
        for i in range(len(s)):
            if(s[i]=='1'):
                oneCount+=1
        for i in range(len(s)-1):
            x = 0
            if(s[i]=='0'):
                zeroCount+=1
                x = oneCount + zeroCount
                if(x>max):
                    max = x
            elif(s[i]=='1'):
                oneCount-=1
                x = oneCount + zeroCount
                if(x>max):
                    max = x
        return max